galasa-ui/src/components/runs/TestRunDetails.tsx

```diff
@@ -71,21 +72,29 @@ const TestRunDetails = ({ runId, runDetails, runLog, runArtifacts }: TestRunDeta
 
     setIsLoading(true);
 
-    try {
-
-      if (runDetails) {
-        extractRunDetails(runDetails);
-        setArtifacts(runArtifacts);
-        setLogs(runLog);
+    const loadUser = async () => {
+      try {
+
+        const runDetails = await runDetailsPromise;
+        const runArtifacts = await runArtifactsPromise;
+        const runLog = await runLogPromise;
+  
+        if (runDetails) {
+          extractRunDetails(runDetails);
+          setArtifacts(runArtifacts);
+          setLogs(runLog);
+        }
+  
+      } catch (err) {
+        setIsError(true);
+      } finally {
+        setIsLoading(false);
       }
+    };
 
-    } catch (err) {
-      setIsError(true);
-    } finally {
-      setIsLoading(false);
-    }
+    loadUser();
 
-  }, [runDetails, runArtifacts, runLog]);
+  }, [runDetailsPromise, runArtifactsPromise, runLogPromise]);
 
   if (isError) {
     return <ErrorPage />;
```

Comment:
Minor - for the test run details page, is there a more suitable loading indicator that we can use instead of the full-page indicator that appears? Maybe like a skeleton that shows text loading?